src/Observable.ts

```diff
@@ -247,7 +247,7 @@ export class Observable<T> implements CoreOperators<T>  {
   publishBehavior: (value: any) => ConnectableObservable<T>;
   publishReplay: (bufferSize?: number, windowTime?: number, scheduler?: Scheduler) => ConnectableObservable<T>;
   publishLast: () => ConnectableObservable<T>;
-  race: (...observables: Array<Observable<any>>) => Observable<any>;
+  race: (...observables: Array<Observable<T>>) => Observable<T>;
```

Comment:
> Is my understanding correct? 

Yes. your understanding is correct. We still have not implemented to accept `Promise<T>` to `race()` yet.
Perhaps, my pull request (and commit message) misleads you easily. Sorry.